internal/auth0/user/resource_permission.go

```diff
@@ -0,0 +1,169 @@
+package user
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/auth0/go-auth0/management"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/auth0/terraform-provider-auth0/internal/mutex"
+)
+
+// NewPermissionResource will return a new auth0_connection_client resource.
+func NewPermissionResource() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"user_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "ID of user to associate the permission to.",
+			},
+			"permission": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of the permission.",
+			},
+			"resource_server_identifier": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Identifier (ID) of the resource server that the permission is associated with.",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Description of the permission.",
+			},
+			"resource_server_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Name of the resource server that the permission is associated with.",
+			},
+		},
+		CreateContext: createUserPermission,
+		ReadContext:   readUserPermission,
+		DeleteContext: deleteUserPermission,
+		Importer: &schema.ResourceImporter{
+			StateContext: importUserPermission,
+		},
+		Description: "With this resource, you can manage user permissions.",
+	}
+}
+
+func createUserPermission(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*management.Management)
+
+	userId := data.Get("user_id").(string)
```

Comment:
```suggestion
	userID := data.Get("user_id").(string)
```

ID, URL, HTTP etc. are special names in golang and should be all capitalized.

Reply:
Can we enforce with linting?

Reply:
Absolutely, feel free to leave as is in this PR if you want and we can have another follow up one with the updates to the golangci-lint.

Reply:
Fixed.